server/sonar-web/src/main/js/apps/issues2/sidebar/AssigneeFacet.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import { sortBy, uniq, without } from 'lodash';
+import FacetBox from './components/FacetBox';
+import FacetHeader from './components/FacetHeader';
+import FacetItem from './components/FacetItem';
+import FacetItemsList from './components/FacetItemsList';
+import type { ReferencedUser } from '../utils';
+import Avatar from '../../../components/ui/Avatar';
+import { translate } from '../../../helpers/l10n';
+
+type Props = {|
+  assigned: boolean,
+  assignees: Array<string>,
+  onChange: (changes: {}) => void,
+  onToggle: (property: string) => void,
+  open: boolean,
+  stats?: { [string]: number },
+  referencedUsers: { [string]: ReferencedUser }
+|};
+
+export default class AssigneeFacet extends React.PureComponent {
+  props: Props;
+
+  static defaultProps = {
+    open: true
+  };
+
+  property = 'assignees';
+
+  handleItemClick = (itemValue: string) => {
+    if (itemValue === '') {
+      // unassigned
+      this.props.onChange({ assigned: !this.props.assigned, assignees: [] });
+    } else {
+      // defined assignee
+      const { assignees } = this.props;
+      const newValue = sortBy(
+        assignees.includes(itemValue)
+          ? without(assignees, itemValue)
+          : uniq([...assignees, itemValue])
```

Comment:
I think there is no need for `uniq` ? Because if the item is already in `modules` we remove it, so I think the `uniq` will never be that useful... If that's really the case this comment probably apply to all facets.